index.js

```diff
@@ -12,7 +12,63 @@ function setMultipart (req, done) {
   done()
 }
 
+function attachToBody (options, req, reply, next) {
+  req[kMultipart] = true
+
+  const body = { }
+  const mp = req.multipart((field, file, filename, encoding, mimetype) => {
+    body[field] = {
+      // stream: file,
+      data: [],
+      filename,
+      encoding,
+      mimetype,
+      limit: false
+    }
+
+    const fileData = []
+    const isCustomDataConsumer = typeof options.onData === 'function'
+    const onData = isCustomDataConsumer
+      ? options.onData
+      : (fieldName, data) => { fileData.push(data) }
+
+    file.on('data', (data) => { onData(field, data) })
+    file.on('limit', () => { body[field].limit = true })
+
+    file.on('end', () => {
+      if (!isCustomDataConsumer) {
+        body[field].data = Buffer.concat(fileData.map((part) => Buffer.isBuffer(part) ? part : Buffer.from(part)))
```

Comment:
The type is not really going to change. If you are reading buffers, you’ll always read buffers. If you are concatenating strings, you should just use +.

Reply:
you can just use concat-stream